packages/containers/src/List/selector.js

```diff
@@ -41,91 +41,74 @@ export function configureGetPagination(state, { collectionId }) {
+		let compare = (a, b) => {
+			if (a.get(sortBy)) {
+				if (a.get(sortBy).localCompare) {
+					return a.get(sortBy).localeCompare(b.get(sortBy));
 				}
-			}
-			return results;
-		},
-	);
-
-	const getSortedList = createSelector(
-		[getFilteredList, getComponentState(localConfig.collectionId)],
-		(items, componentState) => {
-			if (!List.isList(items)) {
-				return new List();
-			}
-			let results = items;
-			if (componentState) {
-				const sortBy = componentState.get('sortOn');
-				const sortAsc = componentState.get('sortAsc');
-				let compare = (a, b) => {
-					if (a.get(sortBy)) {
-						if (a.get(sortBy).localCompare) {
-							return a.get(sortBy).localeCompare(b.get(sortBy));
-						}
-						const aValue = `${a.get(sortBy) || ''}`.toLowerCase();
-						const bValue = `${b.get(sortBy) || ''}`.toLowerCase();
-						if (aValue < bValue) {
-							return -1;
-						}
-						if (aValue > bValue) {
-							return 1;
-						}
-						return 0;
-					}
-					if (!b[sortBy]) {
-						return 0;
-					}
+				const aValue = `${a.get(sortBy) || ''}`.toLowerCase();
+				const bValue = `${b.get(sortBy) || ''}`.toLowerCase();
+				if (aValue < bValue) {
 					return -1;
-				};
-
-				const sortedColumn = get(localConfig, 'columns', [])
-					.filter(column => column.key === sortBy)
-					.pop();
-				if (sortedColumn && sortedColumn.sortFunction) {
-					compare = cmf.registry.getFromRegistry(sortedColumn.sortFunction)(sortBy);
 				}
-				results = results.sort(compare);
-				if (!sortAsc) {
-					results = results.reverse();
+				if (aValue > bValue) {
+					return 1;
 				}
+				return 0;
 			}
-			return results;
-		},
-	);
+			if (!b[sortBy]) {
+				return 0;
+			}
+			return -1;
+		};
+
+		const sortedColumn = get(localConfig, 'columns', [])
+			.filter(column => column.key === sortBy)
+			.pop();
+		if (sortedColumn && sortedColumn.sortFunction) {
+			compare = cmf.registry.getFromRegistry(sortedColumn.sortFunction)(sortBy);
+		}
+		results = results.sort(compare);
+		if (!sortAsc) {
+			results = results.reverse();
+		}
+	}
+	return results;
+}
 
-	return createSelector([getSortedList, getComponentState], items => items);
+export function filterList(componentState, items, localConfig) {
+	let results = items;
+	if (componentState && componentState.get('searchQuery')) {
+		results = items.filter(item =>
+			contains(item, componentState.get('searchQuery'), localConfig.columns),
+		);
+	}
+	return results;
 }
 
 export function configureGetPagedItems(configure) {
-	const getSortedList = configureGetFilteredItems(configure);
-
 	const getPagedList = createSelector(
-		[getSortedList, getComponentState(configure.collectionId)],
-		(items, componentState) => {
-			let results = items;
+		[getComponentState(configure.collectionId)],
+		componentState => {
+			let results = configure.items;
 			if (componentState) {
+				if (componentState.get('items')) {
+					results = componentState && componentState.get('items');
+				}
```

Comment:
The `configureGetPagedItems` could have been refactor but whatever...
However you could have done the following :
```javascript
const stateItems = componentState.get('items');
if (stateItems) {
    results = stateItems;
}
```
and even
```javascript
results = componentState.get('items', results);
```